library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProvider.java

```diff
@@ -205,20 +216,72 @@ void resolveSettingsApi(Status status) {
+                        if (locationAvailability != null && locationAvailability.isLocationAvailable()) {
+                            getSourceProvider().getLastLocation()
+                                    .addOnCompleteListener(new OnCompleteListener<Location>() {
+                                        @Override
+                                        public void onComplete(@NonNull Task<Location> task) {
+                                            /*
+                                             * Returns the best most recent location currently available.
+                                             *
+                                             * If a location is not available, which should happen very rarely, null will be returned.
+                                             * The best accuracy available while respecting the location permissions will be returned.
+                                             *
+                                             * This method provides a simplified way to get location.
+                                             * It is particularly well suited for applications that do not require an accurate location and that do not want to maintain extra logic for location updates.
+                                             *
+                                             * GPS location can be null if GPS is switched off
+                                             */
+                                            if (task.isSuccessful() && task.getResult() != null) {
+                                                Location lastKnownLocation = task.getResult();
+
+                                                LogUtils.logI("LastKnowLocation is available.");
+                                                onLocationChanged(lastKnownLocation);
+
+                                                requestLocation(true);
+                                            } else {
+                                                LogUtils.logI("LastKnowLocation is not available.");
+
+                                                requestLocation(false);
+                                            }
+                                        }
+                                    });
+
+                        } else {
+                            LogUtils.logI("LastKnowLocation is not available.");
+
+                            requestLocation(false);
+                        }
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        LogUtils.logI("LastKnowLocation is not available.");
+
+                        requestLocation(false);
+                    }
+                });
+    }
+
+    void requestLocation(boolean locationIsAlreadyAvailable) {
```

Comment:
Please avoid introducing a new method to request a location update, use either `locationRequired` or `requestLocationUpdate` depending on the conditions.

Reply:
This needed, because checkLastKnowLocation() is task now. Function stores some code from old onConnected(Bundle).

Reply:
I see